libsolidity/Types.cpp

```diff
@@ -323,6 +323,16 @@ TypePointer IntegerConstantType::binaryOperatorResult(Token::Value _operator, Ty
 				return TypePointer();
 			value = m_value % other.m_value;
 			break;
+		case Token::EXP:
+			if (other.m_value < 0)
+				BOOST_THROW_EXCEPTION(InternalCompilerError() << errinfo_comment("exponent can't be negative"));
+			else
+			{
+				value = boost::multiprecision::powm(m_value, other.m_value, bigint(2) << 256);
```

Comment:
@chriseth How about this change? I explicitly check the overflow condition and the negative exponent.